src/mon/OSDMonitor.cc

```diff
@@ -796,14 +796,20 @@ class OSDUtilizationDumper : public CrushTreeDumper::Dumper<F> {
 
   bool get_osd_utilization(int id, int64_t* kb, int64_t* kb_used,
 			   int64_t* kb_avail) const {
+    // we always return true for OSDs as the specific OSDs can be out,
```

Comment:
> But there is no reason we stop to process the whole bucket if any one of the OSDs under this bucket is abnormal

agreed. and could you be specific why checking `osdmap.is_out(id)` in `get_bucket_utilization()`  does not work, or is a bad idea?

we are tackling two cases here:

1. we should not stop calc a bucket's usage when running into an OSD gets marked out: in this case, i'd suggest check `osdmap.is_out(id)` explicitly. actually marking an OSD out will reset its stat to 0, see `PGMonitor::check_osd_map()` and `PGMap::apply_incremental()`
1. in other cases, i just want to be sure it's expected behavior, if it's not, i don't want to cover it up using this fix. if we destroy an OSD, its stat will be removed from `PGMap::osd_stat`, see `PGMonitor::check_osd_map()`, in that case, why is its parent bucket still holding it?
```c++
      if (p->second & CEPH_OSD_EXISTS) {
	// whether it was created *or* destroyed, we can safely drop
	// it's osd_stat_t record.
	dout(10) << __func__ << "  osd." << p->first
	         << " created or destroyed" << dendl;
	pending_inc.rm_stat(p->first);

	// and adjust full, nearfull set
	pg_map.nearfull_osds.erase(p->first);
	pg_map.full_osds.erase(p->first);
      }
```